### PR TITLE
fix(DB/SAI): Mai'Zoth casts Frost Armor

### PR DESCRIPTION
**Pull request by GastonCK**
## Changes Proposed:
-  Mai'Zoth now casts Frost Armor
-  

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/16582
- Closes https://github.com/chromiecraft/chromiecraft/issues/5556

## SOURCE:
Wrath Classic
https://www.wowhead.com/wotlk/spell=12544/frost-armor
https://www.wowhead.com/wotlk/npc=818/maizoth#abilities
https://youtu.be/_Ppcj-ZdT58?t=282

## Tests Performed:
Tested
WoWScrnShot_062123_234124

## How to Test the Changes:

- .go c id 818 he must have frost armor on
- kill and respawn, he must buff himself as soon as he spawns
- purge/dispel and he must rebuff (if enough mana)

## Known Issues and TODO List:

- [ ]
- [ ]

### CODE REVIEW

**Comment by pangolp** [comment]
Thank you for the PR.

File: data/sql/updates/pending_db_world/FixMaiZothFrostArmor.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+UPDATE creature_template SET AIName = 'SmartAI' WHERE entry = 818;
```
Comment (pangolp) [inline data/sql/updates/pending_db_world/FixMaiZothFrostArmor.sql:2]: The NPC already has a smart_scripts, so it makes an update, it is not necessary in this case. This is done, when the npc or whatever is being programmed, still has nothing assigned, but if you are deleting by removing the `id`, that means, that at least the `smart_scripts` has 2 more lines, so the update is unnecessary in this case.

**Comment by pangolp** [comment]
```sql
DELETE FROM `smart_scripts` WHERE `entryorguid`=818 AND `source_type`=0 AND `id`=2;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(818, 0, 2, 0, 23, 0, 100, 0, 12544, 0, 10000, 10000, 0, 11, 12544, 64, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Mai\'Zoth - On Aura \'Frost Armor\' - Cast \'Frost Armor\'');
```

In this way, it does not fail. In case you want to use it.
Welcome to the boat.

**Comment by GastonCK** [comment]
Is it correct now?

**Comment by pangolp** [comment]
> Is it correct now?

I think so, although I have not tested it ingame.
At least the tests were passed.
I don't like to use parentheses in SQL queries.
At least when they are single-valued.
But well, that's just a matter of taste.
I think the result is the same.

I write it this way, but again, the result is the same.
```sql
DELETE FROM `smart_scripts` WHERE `entryorguid`=818 AND `source_type`=0 AND `id`=2;
```

**Review by pangolp — APPROVED** [review]

**Comment by Gultask** [comment]
> I write it this way, but again, the result is the same.

Afaik that's the Keira output, so it's ok

File: data/sql/updates/pending_db_world/FixMaiZothFrostArmor.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 818) AND (`source_type` = 0) AND (`id` IN (2));
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(818, 0, 2, 0, 23, 0, 100, 0, 12544, 0, 10000, 10000, 0, 11, 12544, 64, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'MaiZoth - Missing \'Frost Armor\' - Cast \'Frost Armor\'');
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/FixMaiZothFrostArmor.sql:4]:
```suggestion
DELETE FROM `smart_scripts` WHERE (`entryorguid` = 818) AND (`source_type` = 0) AND (`id` IN (2));
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(818, 0, 2, 0, 23, 0, 100, 0, 12544, 0, 10000, 10000, 0, 11, 12544, 64, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Mai\'Zoth - On Aura \'Frost Armor\' Missing - Cast \'Frost Armor\'');
```

**Review by Gultask — APPROVED** [review]

File: data/sql/updates/pending_db_world/FixMaiZothFrostArmor.sql (hunk 2 above)
Comment (Gultask) [reply data/sql/updates/pending_db_world/FixMaiZothFrostArmor.sql]: I just changed the comment a bit and deleted that empty line, nothing special

**Comment by Gultask** [comment]
Also when available, remember to add the CC issue in the Issues Addressed section as well :)

**Pull request merged**